Scripts/common.ps1

```diff
@@ -0,0 +1,186 @@
+    $ExistingPRs = @($ExistingOpenPRs) + @($ExistingMergedPRs)    
+
+    if ($ExistingPRs.Count -gt 0) {
+        $ExistingPRs | ForEach-Object {
+            Write-Host "Found existing PR: $($_.title)"
+            Write-Host "-> $($_.url)"
+        }
+        exit 0
+    }
+    else {
+
+        return $true
+    }
+}
+
+function Get-VersionAndUrl {
+    param(
+        [Parameter(Mandatory = $false)] [string] $wingetPackage = ${Env:PackageName},
+        [Parameter(Mandatory = $false)] [string] $WebsiteURL = ${Env:WebsiteURL}
+    )
+
+    $scriptPath = ".\Scripts\Update-$($wingetPackage).ps1"
+
+    if (-not (Test-Path -Path $scriptPath)) {
+        Write-Host "The script '$scriptPath' does not exist. Please check the wingetPackage parameter and the current directory."
+        exit 1
+    }
+
+    $Latest = & $scriptPath -WebsiteURL $WebsiteURL -wingetPackage $wingetPackage
+
+
+    if (!($Latest | Get-Member -Name "Version") -and !($Latest | Get-Member -Name "URLs")) {
+
+        $lines = $Latest -split "`n"
+
+        $versionPattern = '^\d+(\.\d+)*$'
+        $urlPattern = '^http[s]?:\/\/[^\s]+$'
+
+        $version = $lines | Where-Object { $_ -match $versionPattern }
+        $URLs = $lines | Where-Object { $_ -match $urlPattern }
+
+        if ($version -and $URLs) {
+            $Latest = @{
+                Version = $version
+                URLs    = $URLs
+            }
+        }
+        else {
+            Write-Host "No Version ($version) or URL ($($URLs -join ',') found."
+            exit 1
+        }
+    }
+
+    Write-Host "Found latest version: $version with URLs: $($Latest.URLs -join ',')"
+    return $Latest
+}
+
+function Update-WingetPackage {
+    param(
+        [Parameter(Mandatory = $true)] [string] $WebsiteURL,
+        [Parameter(Mandatory = $false)] [string] $wingetPackage = ${Env:PackageName},
+        [Parameter(Mandatory = $false)][AllowEmptyString()] [ValidateSet("Komac", "WinGetCreate")] [string] $With = "Komac"
+    )
+    $gitToken = Test-GitHubToken
+
+    $Latest = Get-VersionAndUrl -wingetPackage $wingetPackage -WebsiteURL $WebsiteURL
+
+    $prMessage = "Update version: $wingetPackage version $($Latest.Version)"
+
+    $PackageAndVersionInWinget = Test-PackageAndVersionInGithub -wingetPackage $wingetPackage -latestVersion $($Latest.Version)
+
+
+    if ($PackageAndVersionInWinget) {
+
+        $ExistingPRs = Test-ExistingPRs -wingetPackage $wingetPackage -latestVersion $($Latest.Version)
+        
+        if ($ExistingPRs) {
+            Write-Host "Downloading $with and open PR for $wingetPackage Version $($Latest.Version)"
+            Switch ($with) {
+                "Komac" {
+                    Invoke-WebRequest "https://github.com/russellbanks/Komac/releases/download/v2.2.1/KomacPortable-x64.exe" -OutFile komac.exe
```

Comment:
We should either make the komac version configurable or use the latest.  
As we already use latest for wingetcreate I would prefer the same for komac

Reply:
ja dann mach doch mein Gott